src/main/java/io/appium/java_client/android/ListensToLogcatMessages.java

```diff
@@ -35,11 +37,10 @@
 
     /**
      * Start logcat messages broadcast via web socket.
-     * This method assumes that Appium server is running on localhost and
-     * is assigned to the default port (4723).
      */
     default void startLogcatBroadcast() {
-        startLogcatBroadcast("localhost", DEFAULT_APPIUM_PORT);
+        URL remoteAddress = ((AppiumDriver) this).getRemoteAddress();
+        this.startLogcatBroadcast(remoteAddress.getHost(), remoteAddress.getPort());
```

Comment:
This won't work if the client is connected to a hub. The hub cannot proxy/forward web socket connections

Reply:
A hub?
I looked through the code and the only class that implements the `ListensToLogcatMessages` is `AndroidDriver<T extends WebElement>`. This class on the other hand extends `AppiumDriver`. Every constructor of the AppiumDriver class writes its address to the `remoteAdress` variable, which `getRemoteAddress` returns.
So I don't really see the possibilitiy of it not working. Am I missing something?

Reply:
This endpoint is special from the perspective it uses web socket communication rather than http ones. `remoteAddress` may be pointing to the address of a selenium hub, where no web sockets are running.